### ISSUE TITLE
Cannot add PGP key

### ISSUE THREAD

**Issue by JoyceBabu** [eclipse/packagedrone#105]
I am unable to add PGP key for signing my packages. On clicking the `Add` button, it reloads the page  with the inputs fields pre-filled. There are no error messages.

I tried using

 - `gpg --gen-key` and exporting the key using `gpg --export-secret-key -a <keyid>`
 - online service https://pgpkeygen.com/

for generating the key. But it is not working.

**Comment by ctron** [comment]
Can you maybe provide some screenshot (of course without important information)

**Comment by ctron** [comment]
There is also a short documentation about this in the wiki: https://wiki.eclipse.org/PackageDrone/HowTo/ReverseProxy

**Comment by JoyceBabu** [comment]
> Can you maybe provide some screenshot (of course without important information)
I am attaching the test key generated using https://pgpkeygen.com/ along with the screenshot of the submitted page.

![screen shot 2017-02-27 at 1 37 58 pm](https://cloud.githubusercontent.com/assets/502942/23353379/2bd8d396-fcf2-11e6-8063-a5ab8d3e61a2.png)

[0xDEFB56F9-sec.txt](https://github.com/eclipse/packagedrone/files/802947/0xDEFB56F9-sec.txt)

> There is also a short documentation about this in the wiki: 
> https://wiki.eclipse.org/PackageDrone/HowTo/ReverseProxy

Was that in response to https://github.com/eclipse/packagedrone/issues/104?

**Comment by JoyceBabu** [comment]
I found the following exception

```
packagedrone    | 10:26:28.467 [qtp1036593118-35] DEBUG o.e.p.w.c.v.ControllerValidatorProcessor - Failed to validate
packagedrone    | java.lang.reflect.InvocationTargetException: null
packagedrone    | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_121]
packagedrone    | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_121]
packagedrone    | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_121]
packagedrone    | 	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_121]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.validator.ControllerValidatorProcessor.validate(ControllerValidatorProcessor.java:61) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.validator.CompositeValidator.validate(CompositeValidator.java:45) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.binding.BindingManager.validate(BindingManager.java:280) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.binding.BindingManager.bindProperties(BindingManager.java:270) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.form.FormDataBinder.construct(FormDataBinder.java:146) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.form.FormDataBinder.performBind(FormDataBinder.java:70) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.binding.BindingManager.bindValue(BindingManager.java:79) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.binding.BindingManager.bind(BindingManager.java:113) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerCall.processCall(ControllerCall.java:152) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerCall.lambda$2(ControllerCall.java:110) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerCall.runForward(ControllerCall.java:79) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerCall.lambda$0(ControllerCall.java:65) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.sec.web.controller.SecuredControllerInterceptor.before(SecuredControllerInterceptor.java:56) ~[org.eclipse.packagedrone.sec.web_1.0.0.20160203-1701.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerCall.runForward(ControllerCall.java:62) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerCall.lambda$0(ControllerCall.java:65) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.sec.web.controller.HttpContraintControllerInterceptor.before(HttpContraintControllerInterceptor.java:48) ~[org.eclipse.packagedrone.sec.web_1.0.0.20160203-1701.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerCall.runForward(ControllerCall.java:62) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerCall.call(ControllerCall.java:110) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerEntry.findHandler(ControllerEntry.java:107) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.controller.ControllerTracker.handleRequest(ControllerTracker.java:132) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.DispatcherServlet.mapRequest(DispatcherServlet.java:183) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.DispatcherServlet.service(DispatcherServlet.java:106) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
packagedrone    | 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[javax.servlet_3.1.0.v201410161800.jar:3.1.0]
packagedrone    | 	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:837) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1689) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.doRunFilter(FilterTracker.java:134) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.access$1(FilterTracker.java:117) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker$2.doFilter(FilterTracker.java:127) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.repo.web.googlebot.GoogleBotFilter.doFilter(GoogleBotFilter.java:50) ~[na:na]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker$TrackedFilter.doFilter(FilterTracker.java:51) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.doRunFilter(FilterTracker.java:122) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.access$1(FilterTracker.java:117) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker$2.doFilter(FilterTracker.java:127) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.sec.web.filter.SecurityFilter.processFilter(SecurityFilter.java:102) ~[org.eclipse.packagedrone.sec.web_1.0.0.20160203-1701.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.sec.web.filter.SecurityFilter.doFilter(SecurityFilter.java:75) ~[org.eclipse.packagedrone.sec.web_1.0.0.20160203-1701.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker$TrackedFilter.doFilter(FilterTracker.java:51) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.doRunFilter(FilterTracker.java:122) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.doFilter(FilterTracker.java:114) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
packagedrone    | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1676) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:581) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:70) ~[org.ops4j.pax.web.pax-web-jetty_4.2.5.jar:na]
packagedrone    | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548) ~[org.eclipse.jetty.security_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:271) ~[org.ops4j.pax.web.pax-web-jetty_4.2.5.jar:na]
packagedrone    | 	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80) ~[org.ops4j.pax.web.pax-web-jetty_4.2.5.jar:na]
packagedrone    | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.server.Server.handle(Server.java:524) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:319) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:253) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:273) ~[org.eclipse.jetty.io_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95) ~[org.eclipse.jetty.io_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93) ~[org.eclipse.jetty.io_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
packagedrone    | 	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_121]
packagedrone    | Caused by: java.lang.NoClassDefFoundError: org/bouncycastle/openpgp/operator/PGPDigestCalculatorProvider
packagedrone    | 	at org.eclipse.packagedrone.repo.signing.pgp.web.managed.ServiceController.validateKey(ServiceController.java:118) ~[na:na]
packagedrone    | 	... 68 common frames omitted
packagedrone    | Caused by: java.lang.ClassNotFoundException: org.bouncycastle.openpgp.operator.PGPDigestCalculatorProvider cannot be found by org.eclipse.packagedrone.utils.security_1.0.0.20160630-0906
packagedrone    | 	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:439) ~[org.eclipse.osgi_3.10.100.v20150529-1857.jar:na]
packagedrone    | 	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:352) ~[org.eclipse.osgi_3.10.100.v20150529-1857.jar:na]
packagedrone    | 	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:344) ~[org.eclipse.osgi_3.10.100.v20150529-1857.jar:na]
packagedrone    | 	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:160) ~[org.eclipse.osgi_3.10.100.v20150529-1857.jar:na]
packagedrone    | 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_121]
packagedrone    | 	... 69 common frames omitted

```

**Comment by alexanderBendo** [comment]
Same problem here with org.eclipse.packagedrone.server-0.14.0-3 and CentOS 7

```
07:12:05.034 [qtp486453355-60] DEBUG o.e.p.w.c.v.ControllerValidatorProcessor - Failed to validate
java.lang.reflect.InvocationTargetException: null
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_121]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_121]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_121]
	at org.eclipse.packagedrone.web.controller.validator.ControllerValidatorProcessor.validate(ControllerValidatorProcessor.java:61) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.validator.CompositeValidator.validate(CompositeValidator.java:45) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.binding.BindingManager.validate(BindingManager.java:280) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.binding.BindingManager.bindProperties(BindingManager.java:270) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.form.FormDataBinder.construct(FormDataBinder.java:146) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.form.FormDataBinder.performBind(FormDataBinder.java:70) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.binding.BindingManager.bindValue(BindingManager.java:79) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.binding.BindingManager.bind(BindingManager.java:113) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerCall.processCall(ControllerCall.java:152) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerCall.lambda$2(ControllerCall.java:110) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerCall.runForward(ControllerCall.java:79) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerCall.lambda$0(ControllerCall.java:65) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.sec.web.controller.SecuredControllerInterceptor.before(SecuredControllerInterceptor.java:56) ~[org.eclipse.packagedrone.sec.web_1.0.0.20160203-1701.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerCall.runForward(ControllerCall.java:62) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerCall.lambda$0(ControllerCall.java:65) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.sec.web.controller.HttpContraintControllerInterceptor.before(HttpContraintControllerInterceptor.java:48) ~[org.eclipse.packagedrone.sec.web_1.0.0.20160203-1701.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerCall.runForward(ControllerCall.java:62) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerCall.call(ControllerCall.java:110) [org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerEntry.findHandler(ControllerEntry.java:107) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.controller.ControllerTracker.handleRequest(ControllerTracker.java:132) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.DispatcherServlet.mapRequest(DispatcherServlet.java:183) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at org.eclipse.packagedrone.web.DispatcherServlet.service(DispatcherServlet.java:106) ~[org.eclipse.packagedrone.web_1.0.0.20160623-1148.jar:na]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[javax.servlet_3.1.0.v201410161800.jar:3.1.0]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:837) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1689) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.doRunFilter(FilterTracker.java:134) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.access$1(FilterTracker.java:117) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker$2.doFilter(FilterTracker.java:127) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.packagedrone.repo.web.googlebot.GoogleBotFilter.doFilter(GoogleBotFilter.java:50) ~[na:na]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker$TrackedFilter.doFilter(FilterTracker.java:51) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.doRunFilter(FilterTracker.java:122) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.access$1(FilterTracker.java:117) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker$2.doFilter(FilterTracker.java:127) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.packagedrone.sec.web.filter.SecurityFilter.processFilter(SecurityFilter.java:102) ~[org.eclipse.packagedrone.sec.web_1.0.0.20160203-1701.jar:na]
	at org.eclipse.packagedrone.sec.web.filter.SecurityFilter.doFilter(SecurityFilter.java:75) ~[org.eclipse.packagedrone.sec.web_1.0.0.20160203-1701.jar:na]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker$TrackedFilter.doFilter(FilterTracker.java:51) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.doRunFilter(FilterTracker.java:122) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.packagedrone.web.dispatcher.internal.FilterTracker.doFilter(FilterTracker.java:114) ~[org.eclipse.packagedrone.web.dispatcher_1.0.0.20160203-1224.jar:na]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1676) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:581) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:70) ~[org.ops4j.pax.web.pax-web-jetty_4.2.5.jar:na]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548) ~[org.eclipse.jetty.security_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:271) ~[org.ops4j.pax.web.pax-web-jetty_4.2.5.jar:na]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511) ~[org.eclipse.jetty.servlet_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80) ~[org.ops4j.pax.web.pax-web-jetty_4.2.5.jar:na]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.server.Server.handle(Server.java:524) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:319) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:253) ~[org.eclipse.jetty.server_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:273) ~[org.eclipse.jetty.io_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95) ~[org.eclipse.jetty.io_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93) ~[org.eclipse.jetty.io_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589) ~[org.eclipse.jetty.util_9.3.10.v20160621.jar:9.3.10.v20160621]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_121]
Caused by: java.lang.NoClassDefFoundError: org/bouncycastle/openpgp/operator/PGPDigestCalculatorProvider
	at org.eclipse.packagedrone.repo.signing.pgp.web.managed.ServiceController.validateKey(ServiceController.java:118) ~[na:na]
	... 68 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.bouncycastle.openpgp.operator.PGPDigestCalculatorProvider cannot be found by org.eclipse.packagedrone.utils.security_1.0.0.20160630-0906
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:439) ~[org.eclipse.osgi_3.10.100.v20150529-1857.jar:na]
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:352) ~[org.eclipse.osgi_3.10.100.v20150529-1857.jar:na]
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:344) ~[org.eclipse.osgi_3.10.100.v20150529-1857.jar:na]
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:160) ~[org.eclipse.osgi_3.10.100.v20150529-1857.jar:na]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_121]
	... 69 common frames omitted
```

Looks like a problem with BouncyCastle so I installed packages `bouncycastle-1.50-1` and `bouncycastle-pkix-1.50-1`, but it didn't solve the problem.

**Comment by alexanderBendo** [comment]
Just tried with Oracle Java and it doesn't work either, same exception
```
# java -version
java version "1.8.0_101"
Java(TM) SE Runtime Environment (build 1.8.0_101-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.101-b13, mixed mode)
```

```
Caused by: java.lang.ClassNotFoundException: org.bouncycastle.openpgp.operator.PGPDigestCalculatorProvider cannot be found by org.eclipse.packagedrone.utils.security_1.0
.0.20160630-0906  
```

**Comment by alexanderBendo** [comment]
Tried with Ubuntu, same error.

**Comment by ctron** [comment]
I am sorry but I currently I don't have to debug this.

**Comment by alexanderBendo** [comment]
This issue is solved in version 0.14.1.20170407-0757. I was able to add a managed key and sign my packages. 😄 

Thank you sooo much 🙇

**Comment by ctron** [comment]
Thanks for the feedback.
